Finished BankNumber Algorithm and Login/Register Styling + small changes

Chenentano opened the pull request:
Check commit msgs

Chenentano committed:
Update validation on account fields and adjust home page content

In this update, changes were made both to the frontend and backend code. The account number placeholder text on the homepage was modified. In the backend, additional Java validation annotations were added to the Account entity to ensure valid email input and proper account names by the users.

barbarahellmann reviewed (approved):
Im Frontend hast du das Styling im classname. Dass macht den Code für mich unübersichtlicher.

Chenentano committed:
Add functionality to generate bank account numbers

A new feature has been implemented to generate unique bank account numbers for accounts. If accounts are already existing and do not have an assigned bank account number, this will be generated and updated in their information. The frontend code has also been adjusted to correctly display the generated account numbers.

Chenentano committed:
Update user authentication and improve UI design

Revamped the user authentication process on Register and Login pages by storing more user-related data in local storage and redirecting to the homepage upon successful authentication. Update the UI design of these pages for better user experience, including changes in background, text style, and element placement.
